src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -190,13 +193,15 @@ class GoogleExtensions(
 
       // make all the publish messages for the previously synced groups
       messages <- previouslySyncedIds.traverse {
-          // if it's a group that isn't an access policy
-          case groupName: WorkbenchGroupName => makeGroupPublishMessages(groupName)
-          // if it's the access policy of a managed group
-          case FullyQualifiedPolicyId(FullyQualifiedResourceId(ManagedGroupService.managedGroupTypeName, id: ResourceId),
-                                                 ManagedGroupService.adminPolicyName | ManagedGroupService.memberPolicyName) =>
-            makeGroupPublishMessages(WorkbenchGroupName(id.value))
-          // if it's an access policy on a resource that's not a managed group
+          // it is a group that isn't an access policy, could be a managed group
+          case groupName: WorkbenchGroupName =>
+            makeConstrainedResourceAccessPolicyMessages(groupName).map(_  :+ groupName.toJson.compactPrint)
+
+          // it is the admin or member access policy of a managed group
+          case accessPolicyId@FullyQualifiedPolicyId(FullyQualifiedResourceId(ManagedGroupService.managedGroupTypeName, id: ResourceId), ManagedGroupService.adminPolicyName | ManagedGroupService.memberPolicyName) =>
```

Comment:
Is there any way we can simplify this line or break it up or something?

Reply:
I made it  a little shorter but it is a pattern match so I don't know what else to do